modules/terminate-agent-hook/lambda/lambda_function.py

```diff
@@ -87,13 +90,67 @@ def ec2_list(client, **args):
 
     return _terminate_list
 
+
+def remove_unused_ssh_key_pairs(client, executor_name_part):
+    print(json.dumps({
+        "Level": "info",
+        "Message": f"Removing unused SSH key pairs for agent {executor_name_part}"
+    }))
+
+    reservations = client.describe_instances(Filters=[
```

Comment:
Quick quesiton. I am not used to the AWS python sdk. But for example in typescript / javascript API's are paged. So the calls in the function are most like also paged and only returning the first 50 objects. Is this the case?

Reply:
Oh yes, true. Pagination is done for every 1.000 instances. Guess it's unlikely that you have so many machines running at the same time. But better to fix it now.